tools/translation/hashes.py

```
@@ -0,0 +1,80 @@
+    if not blocks:
+        return body
+    return json.dumps(static_body_segments(body, blocks), ensure_ascii=False, separators=(",", ":"))
+
+
+def translatable_body_hash(frontmatter: str, body: str) -> str:
+    return source_body_hash(translatable_body(frontmatter, body))
+
+
+def legacy_source_hash(frontmatter: str, body: str) -> str:
+    translation_id = read_scalar(frontmatter, "translation_id") or ""
+    payload = f"translation_id={translation_id}\n\n{body}".encode("utf-8")
```

Ruff (0.15.15)

[warning] Unnecessary UTF-8 `encoding` argument to `encode`

Remove unnecessary `encoding` argument

(UP012)